File: facade_pattern/fd.kt
class CPU {
  fun process() {
    println("CPU processing...")
  }
}

class Memory {
  fun load() {
    println("Memory loading...")
  }
}

class SSD {
  fun read() {
    println("SSD reading...")
  }
}

class Computer {
  private val cpu = CPU()
  private val memory = Memory()
  private val ssd = SSD()

  // Facade method
  fun boost() {
    println("Computer starting up...")
    cpu.process()
    memory.load()
    ssd.read()
    println("Computer ready to use!")
  }
}

fun main() {
  val computer = Computer()
  computer.boost()
}

File: facade_pattern/fd.cpp
#include <iostream>
using namespace std;

class CPU {
public:
  void process() {
    cout << "CPU processing..." << endl;
  }
};

class Memory {
public:
  void load() {
    cout << "Memory loading..." << endl;
  }
};

class SSD {
public:
  void read() {
    cout << "SSD reading..." << endl;
  }
};

class Computer {
private:
  CPU cpu;
  Memory memory;
  SSD ssd;

public:
  // Facade method
  void boost() {
    cout << "Computer starting up..." << endl;
    cpu.process();
    memory.load();
    ssd.read();
    cout << "Computer ready to use!" << endl;
  }
};

int main() {
  Computer c;
  c.boost();
  return 0;
}